examples/server/server.cpp

```diff
@@ -2163,16 +2213,19 @@ struct server_context {
                                 GGML_ASSERT(slot.n_prompt_tokens < slot.n_ctx);
                             }
 
-                            common_sampler_reset(slot.smpl);
+                            // Should this be (re-)moved?
+                            //common_sampler_reset(slot.smpl);
 
                             if (slot.params.cache_prompt) {
                                 // reuse any previously computed tokens that are common with the new prompt
                                 slot.n_past = longest_common_prefix(slot.cache_tokens, prompt_tokens);
+                                // Not sure if the for loop below should happen in multiple places but for now I moved it
+                                // until after the entire prompt is processed so that sampling would happen consistently.
 
                                 // push the prompt into the sampling context (do not apply grammar)
-                                for (int i = 0; i < slot.n_past; ++i) {
-                                    common_sampler_accept(slot.smpl, slot.cache_tokens[i], false);
-                                }
+                                // for (int i = 0; i < slot.n_past; ++i) {
+                                //     common_sampler_accept(slot.smpl, slot.cache_tokens[i], false);
+                                // }
```

Comment:
I added a line to print `prompt_tokens.size()` in this part:

```
                    // entire prompt has been processed
                    if (slot.n_past == slot.n_prompt_tokens) {
                        slot.state = SLOT_STATE_DONE_PROMPT;

                        GGML_ASSERT(batch.n_tokens > 0);

                        common_sampler_reset(slot.smpl);

                        // Process all prompt tokens through sampler system
                        for (int i = 0; i < slot.n_prompt_tokens; ++i) {
                            common_sampler_accept(slot.smpl, prompt_tokens[i], false);
                        }

                        // extract the logits only for the last token
                        batch.logits[batch.n_tokens - 1] = true;

                        slot.n_decoded = 0;
                        slot.i_batch   = batch.n_tokens - 1;

                        SLT_INF(slot, "size of prompt_tokens array = %ld\n", prompt_tokens.size());
                        SLT_INF(slot, "prompt done, n_past = %d, n_tokens = %d\n", slot.n_past, batch.n_tokens);
                    }
```
And got this output on my second generation with `'cache_prompt': true` using a decently large context:

```
slot launch_slot_: id  0 | task 215 | processing task
slot update_slots: id  0 | task 215 | tokenizing prompt, len = 233
slot update_slots: id  0 | task 215 | prompt tokenized, n_ctx_slot = 32000, n_keep = 0, n_prompt_tokens = 9483
slot update_slots: id  0 | task 215 | kv cache rm [8816, end)
slot update_slots: id  0 | task 215 | prompt processing progress, n_past = 9483, n_tokens = 667, progress = 0.070336
slot update_slots: id  0 | task 215 | size of prompt_tokens array = 9483
slot update_slots: id  0 | task 215 | prompt done, n_past = 9483, n_tokens = 667

---- DRY sampling debug info:
  Applied penalty 0.8000 to token 1395 ( is) (repeat length 2)
  Applied penalty 0.8000 to token 1681 ('s) (repeat length 2)
  Applied penalty 0.8000 to token 10637 ( looks) (repeat length 2)
  Applied penalty 0.8000 to token 45493 ( beams) (repeat length 2)
  Applied penalty 0.8000 to token 86372 ( nods) (repeat length 2)
  Applied penalty 0.8000 to token 92156 ( shakes) (repeat length 2)
Total DRY penalties applied: 6
DRY Ring Buffer Content (showing 9485 out of 9485 tokens):
```

Both `prompt_tokens.size()` and my ring buffer `last_tokens` are showing the full size of the context (with `last_tokens` being larger by 2 on the last line because my debugging doesn't output until DRY is actually triggered).

I'm not sure if this is a bug that this works, but it seems to be working? I also have a full print out in my debugging code that shows the entirety of the prompt being present, token by token, for all 9483 tokens.